src/server/handlers/launchpad.js

```diff
@@ -310,30 +316,27 @@ export const findSnap = (req, res) => {
         }
       });
     })
-    .catch((error) => responseError(res, error));
+    .catch((error) => sendError(res, error));
 };
 
-export const completeSnapAuthorization = async (req, res) => {
+// Not a route handler, but kept here so that the beginAuthorization and
+// completeAuthorization calls are close together.  Returns a Promise.
+export const completeSnapAuthorization = (session, repositoryUrl,
+                                          dischargeMacaroon) => {
   let snapUrl;
-  checkAdminPermissions(req)
-    .then(() => internalFindSnap(req.body.repository_url))
+  return checkAdminPermissions(session, repositoryUrl)
+    .then(() => internalFindSnap(repositoryUrl))
     .then((result) => {
       snapUrl = result;
       return getLaunchpad().named_post(snapUrl, 'completeAuthorization', {
-        parameters: { discharge_macaroon: req.body.discharge_macaroon },
+        parameters: { discharge_macaroon: dischargeMacaroon },
       });
     })
-    .then(() => {
-      logger.info(`Completed authorization of ${snapUrl}`);
-      return res.status(200).send({
-        status: 'success',
-        payload: {
-          code: 'snap-authorized',
-          message: snapUrl
-        }
-      });
-    })
-    .catch((error) => responseError(res, error));
+    .then(() => logger.info(`Completed authorization of ${snapUrl}`))
+    .catch((error) => {
+      return prepareError(error)
```

Comment:
I hoped it would be not necessary to call `prepareError` outside of `sendError`, but as `completeSnapAuthorization` is called from different part of flow it doesn't actually use `sendError` and needs special treatment, right?